dialogue-core/src/main/java/com/palantir/dialogue/core/QueuedChannel.java

```diff
@@ -165,13 +165,17 @@ private boolean scheduleNextTask() {
 
         @Override
         public void onSuccess(Response result) {
-            response.set(result);
+            if (!response.set(result)) {
```

Comment:
time for an error-prone rule to flag this kind of thing to us?

Reply:
Ya, I think so. It's a super common mistake, so rollout will be difficult more broadly.

I think we can suggest a fix of either:
`Preconditions.checkState(settableFuture.set[Exception](value), "failed to complete")`
or
```java
if (!settableFuture.set[Exception](value)) {
    log.warn("Failed to complete future", value); // requires a logger to be defined, otherwise the check becomes much more complicated.
}
```

Reply:
even if it's hard to roll out everywhere, I think I'd be happy to have it just in this repo to avoid our own foot guns!